src/main/kotlin/eda/shoppingBasket/service/ShoppingBasketService.kt

```diff
@@ -0,0 +1,152 @@
+            //add the shopping basket to the shopping basket repo
+            shoppingBasketRepository.save(shoppingBasket)
+            return shoppingBasketMapper.toDTO(shoppingBasket)
+        }
+        return null
+    }
+
+    fun removeItemFromShoppingBasket(shoppingBasketID: UUID, shoppingBasketItemID: UUID): ShoppingBasketDTO? {
+        val shoppingBasket = shoppingBasketRepository.findById(shoppingBasketID).get()
+        if (shoppingBasket == null) {
+            //maybe return something like:
+            throw Exception("Shopping basket with id $shoppingBasketID does not exist.")
+        }
+        else if (shoppingBasket.shoppingBasketItems.isEmpty()) {
+            //maybe return something like:
+            throw Exception("Shopping basket with id $shoppingBasketID is empty.")
+        }
+        else if (shoppingBasket.shoppingBasketItems.none { it.shoppingBasketItemID == shoppingBasketItemID }) {
+            //maybe return something like:
+            throw Exception("Shopping basket with id $shoppingBasketID does not contain an item with id $shoppingBasketItemID.")
+        }
+        else {
+            val shoppingBasketItem = shoppingBasket.shoppingBasketItems.find { it.shoppingBasketItemID == shoppingBasketItemID }
+            shoppingBasket.shoppingBasketItems.remove(shoppingBasketItem)
+            //i am also removing the shopping basket item from the shopping basket item repo here, not sure if this is what we need rn
+            shoppingBasketItemRepository.deleteById(shoppingBasketItemID)
+            shoppingBasketRepository.save(shoppingBasket)
+            return shoppingBasketMapper.toDTO(shoppingBasket)
+        }
+        return null
+    }
+
+    fun modifyItemQuantity(shoppingBasketID: UUID, shoppingBasketItemID: UUID, newQuantity: Int): ShoppingBasketDTO {
+        val shoppingBasket = shoppingBasketRepository.findById(shoppingBasketID).get()
+        if (shoppingBasket == null) {
+            //maybe return something like:
+            throw Exception("Shopping basket with id $shoppingBasketID does not exist.")
+        }
+        else if (shoppingBasket.shoppingBasketItems.isEmpty()) {
+            //maybe return something like:
+            throw Exception("Shopping basket with id $shoppingBasketID is empty.")
+        }
+        else if (shoppingBasket.shoppingBasketItems.none { it.shoppingBasketItemID == shoppingBasketItemID }) {
+            //maybe return something like:
+            throw Exception("Shopping basket with id $shoppingBasketID does not contain an item with id $shoppingBasketItemID.")
+        }
+        else {
+            val shoppingBasketItem = shoppingBasket.shoppingBasketItems.find { it.shoppingBasketItemID == shoppingBasketItemID }
+            shoppingBasketItem!!.quantity = newQuantity
+            shoppingBasketItemRepository.save(shoppingBasketItem)
+            //updating the shopping basket item repo here too; not sure if this is what we need rn
```

Comment:
we should instead set the cascade type to remove in the spring annotation for the entity

Reply:
But for offering it would be detach (I hope that works that way).

Reply:
> But for offering it would be detach (I hope that works that way).
we are not updating (or supposed to update) our offering repo here, right?